src/Compilers/Core/Portable/PEWriter/Members.cs

```diff
@@ -979,5 +979,33 @@ internal static bool HasBody(this IMethodDefinition methodDef)
             return !methodDef.IsAbstract && !methodDef.IsExternal &&
                 (methodDef.ContainingTypeDefinition == null || !methodDef.ContainingTypeDefinition.IsComObject);
         }
+
+        /// <summary>
+        /// When emitting ref assemblies, some method definitions will not be included.
+        /// </summary>
+        public static bool ShouldInclude(this IMethodDefinition method, EmitContext context)
+        {
+            if (method.IsVirtual)
+            {
+                return true;
+            }
+
+            return ShouldInclude((ITypeDefinitionMember)method, context);
```

Comment:
>return ShouldInclude((ITypeDefinitionMember)method, context); [](start = 12, length = 61)

This feels somewhat strange. I would expect the more general method (the one below) to delegate to more specific method (this one), not the other way around.